packages/expo-ui/ios/ListItem.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import SwiftUI
+
+internal class SwipeAction: Record, Identifiable {
+  required init() { }
+  @Field var id: String
+  @Field var props: ButtonProps
+}
+
+class ListItemViewProps: ExpoSwiftUI.ViewProps {
+  @Field var leadingActions: [SwipeAction]
+  @Field var allowsFullSwipeLeading = true
+
+  @Field var trailingActions: [SwipeAction]
+  @Field var allowsFullSwipeTrailing = true
+  @Field var hideTrailingActionsInEditMode = true
+
+  var onActionPressed = EventDispatcher()
+}
+
+struct ActionButtons: View {
+  let fromArray: [SwipeAction]?
+  let handleButtonPress: (_ id: String) -> Void
+
+  var body: some View {
+    ForEach(fromArray ?? []) { elem in
+      elem.props.onButtonPressed.onEventSent = { _ in
+        handleButtonPress(elem.id)
+      }
+
+      return ExpoUI.Button().environmentObject(elem.props)
+    }
+  }
+}
+
+struct ListItemView: ExpoSwiftUI.View {
+  @Environment(\.editMode) private var editMode
+  @EnvironmentObject var props: ListItemViewProps
+
+  var body: some View {
+    UnwrappedChildren { child, isHostingView in
+      child
+        .if(!isHostingView) {
+          $0
+            .offset(
+              x: UIDevice.current.userInterfaceIdiom == .pad
+              ? IPAD_OFFSET : IPHONE_OFFSET)
+        }
+    }
+    .swipeActions(edge: .leading, allowsFullSwipe: props.allowsFullSwipeLeading) {
+      ActionButtons(fromArray: props.leadingActions) { id in
+        props.onActionPressed.callAsFunction(["id": id])
+      }
+    }
+    .swipeActions(edge: .trailing, allowsFullSwipe: props.allowsFullSwipeTrailing) {
+      if (editMode?.wrappedValue == .inactive || !props.hideTrailingActionsInEditMode) {
```

Comment:
I tried to use the `.if` modifier, but opening edit mode changed the animation when toggling edit mode

https://github.com/user-attachments/assets/6b2053bf-ff80-4781-ba5d-4b422dbfed66

```swift
.if(editMode?.wrappedValue == .inactive || !props.hideTrailingActionsInEditMode) {
  $0.swipeActions(edge: .trailing, allowsFullSwipe: props.allowsFullSwipeTrailing) {
      ActionButtons(fromArray: props.trailingActions) { id in
        props.onActionPressed.callAsFunction(["id": id])
      }
  }
}
```